Follow-ups from `mute user` backend commit merge.

abhijeetbodas2001 opened the pull request:
This contains follow-up work from 3bfcaa39689c9996cb7eb0ec65bd37f727074f6e (#16915)

> Link to /help/ for muted users from API docs for the feature, both the main docs and the events.

The `/help` docs will be added in the commit which enables muting/unmuting from the UI. I'll add these links in that commit when I update it.

> We should change the do_ functions to record entries in RealmAuditLog when muting another user. I think modified_user needs to be the current user, as well as acting_user, so probably we need to store the other user in extra_data (no need to record timestamp, since that's there anyway).

This PR doesn't handle this yet. I didn't understand this. Should I create a new event-type in `AbstractRealmAuditLog`? Would `USER_MUTED` be a fine name or maybe `MUTED_USERS_CHANGED`? And what should be the number-key for it?

> For unmute_user, would it be cleaner to pass the MutedUser object from the user_is_muted check to the remove_user_mute function? This would avoid a second fetch.

Done.

> The backend tests should be tweaked to (1) do everything via the API, rather than calling do_ functions directly -- it's better to have tests be more end-to-end, and (2) do all the repeated times in variables. (I changed a few bits in this direction before merging, but decided to leave the rest to you).

Done!

timabbott commented on zerver/lib/user_mutes.py:
```diff
@@ -29,9 +29,11 @@ def add_user_mute(
-
-def user_is_muted(user_profile: UserProfile, muted_user: UserProfile) -> bool:
-    return MutedUser.objects.filter(user_profile=user_profile, muted_user=muted_user).exists()
+    return mute_object
```
This would be cleaner with a return in both code paths.

abhijeetbodas2001 replied:
Yes, definitely cleaner! Made this change.

abhijeetbodas2001 commented:
@timabbott I've addressed your comment above, and added one more commit which adds tests for the `import` code you had added while merging 3bfcaa39689c9996cb7eb0ec65bd37f727074f6e.

timabbott commented:
Nice, merged, thanks @abhijeetbodas2001!

timabbott commented:
> This PR doesn't handle this yet. I didn't understand this. Should I create a new event-type in AbstractRealmAuditLog? Would USER_MUTED be a fine name or maybe MUTED_USERS_CHANGED? And what should be the number-key for it?

Yep!  I think we can go with `USER_MUTED = 350`

abhijeetbodas2001 commented:
Thanks, I'll make a commit for it when working on the other backend commits from the main PR.
